plugins/nextcloud-rule-exclusions-before.conf

```diff
@@ -739,15 +792,38 @@ SecRule REQUEST_FILENAME "@endsWith /login" \
             ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:pass1-text,\
             ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:pass2"
 
-# Logout token
-SecRule REQUEST_FILENAME "@endsWith /logout" \
-    "id:9508420,\
+# Password reset used in newer version of Nextcloud
+SecRule REQUEST_FILENAME "@rx /lostpassword/set/[^/]+/[^/]+$" \
+    "id:9508411,\
     phase:1,\
     pass,\
     t:none,\
     nolog,\
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
-    ctl:ruleRemoveTargetById=941120;ARGS:requesttoken"
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.password"
+
+# requesttoken is used when logging in or when authenticating via a public share
+# Matches:
+# /login
+# /s/share-id/authenticate/showShare
+# /login/v2/grant
+# /logout
+SecRule REQUEST_FILENAME "@rx /(?:login(?:/v[0-9\.]+/grant)?|s/[^/]+/authenticate/showShare|logout)$" \
+    "id:9508420,\
+    phase:2,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
+    chain"
+    SecRule ARGS:requesttoken "@rx ^[a-zA-Z0-9+=:/]+$" \
+        "t:none,\
+        ctl:ruleRemoveTargetById=920273;ARGS:requesttoken,\
+        ctl:ruleRemoveTargetById=932236;ARGS:requesttoken,\
+        ctl:ruleRemoveTargetById=941100;ARGS:requesttoken,\
+        ctl:ruleRemoveTargetById=941120;ARGS:requesttoken,\
+        ctl:ruleRemoveTargetById=942432;ARGS:requesttoken,\
+        ctl:ruleRemoveTargetById=942450;ARGS:requesttoken"
 
 # Logging in with webauthn
```

Comment:
This looks like a duplicate of the new rule 9508402.

Reply:
it's similar, but it checks the response
I'll add a comment to make what it does clearer

Reply:
Ah, ok.